GDJS/tests/tests/hot-reloader.js

```diff
@@ -36,6 +36,412 @@ describe('gdjs.HotReloader.deepEqual', () => {
+    ambientLightColorB: 127,
+    ambientLightColorG: 127,
+    isLightingLayer: false,
+    followBaseLayerCamera: false,
+  };
+
+  /**
+   * Create and return a minimum working sprite object data.
+   * @internal
+   * @param {{name: string, image: string}} data
+   * @returns {gdjs.SpriteObjectData}
+   */
+  const createSpriteData = ({ name, image }) => {
+    return {
+      type: 'Sprite',
+      name,
+      behaviors: [],
+      variables: [],
+      effects: [],
+      animations: [
+        {
+          name: 'animation',
+          directions: [
+            {
+              sprites: [
+                {
+                  originPoint: { name: 'Origin', x: 0, y: 0 },
+                  centerPoint: {
+                    name: 'Center',
+                    x: 0,
+                    y: 0,
+                    automatic: true,
+                  },
+                  points: [],
+                  hasCustomCollisionMask: false,
+                  customCollisionMask: [],
+                  image,
+                },
+              ],
+              timeBetweenFrames: 0,
+              looping: false,
+            },
+          ],
+          useMultipleDirections: false,
+        },
+      ],
+      updateIfNotVisible: false,
+    };
+  };
+
+  /**
+   * Create and return a minimum working scene data.
+   * @internal
+   * @param {{instances?: Partial<InstanceData>[], objects?: ObjectData[]}} data
+   * @returns {LayoutData}
+   */
+  const createSceneData = ({ instances, objects }) => {
+    return {
+      layers: [baseLayer],
+      variables: [],
+      r: 0,
+      v: 0,
+      b: 0,
+      mangledName: 'Scene1',
+      name: 'Scene1',
+      stopSoundsOnStartup: false,
+      title: '',
+      behaviorsSharedData: [],
+      objects: objects || [
+        createSpriteData({ name: 'MyObject', image: 'ResourceA' }),
+      ],
+      instances: instances
+        ? instances.map((instance) => ({ ...defaultInstance, ...instance }))
+        : [],
+      usedResources: [],
+    };
+  };
+
+  /**
+   * Create and return a minimum working scene data.
+   * @internal
+   * @param {{name: string, instances?: Partial<InstanceData>[], objects?: ObjectData[]}} data
+   * @returns {EventsBasedObjectData}
+   */
+  const createEventsBasedObjectData = ({ name, instances, objects }) => {
+    return {
+      name,
+      variables: [],
+      instances: instances
+        ? instances.map((instance) => ({
+            ...defaultChildInstance,
+            ...instance,
+          }))
+        : [],
+      objects: objects || [
+        createSpriteData({ name: 'MyChildObject', image: 'ResourceA' }),
+      ],
+      layers: [baseLayer],
+      areaMinX: 0,
+      areaMinY: 0,
+      areaMinZ: 0,
+      areaMaxX: 0,
+      areaMaxY: 0,
+      areaMaxZ: 0,
+      defaultSize: null,
+    };
+  };
+
+  /**
+   * @internal
+   * @param {gdjs.RuntimeObject} instance
+   * @returns {string | null}
+   */
+  const getSpriteImage = (instance) => {
```

Comment:
Nitpicking, so that we don't need to read the code to be sure what you mean by a "Sprite image":
```suggestion
  const getSpriteCurrentFrameImage = (instance) => {
```